project/driver_controls/src/driver_io.c

```diff
@@ -0,0 +1,113 @@
+#include <stddef.h>
+
+#include "driver_io.h"
+#include "gpio_it.h"
+#include "input_interrupt.h"
+
+static const DriverIO driver_io[NUM_DRIVER_IO_INPUTS] = {
```

Comment:
The point of the array was to have the ISRs depend on the calling device rather than the GPIO pin. This way, if we need to change pin configurations, it can be done without changing other parts of the code.